chaos_genius/alerts/base_alert_digests.py

```diff
@@ -31,90 +27,107 @@
 
 
 class AlertDigestController:
+    """Controller for anomaly alert digests."""
+
     def __init__(self, frequency: str):
+        """Initializes an anomaly alert digests controller.
 
+        Note: an AlertDigestController instance must only be used for one check/trigger
+        of a digest. The same object must not be re-used.
+
+        Arguments:
+            frequency: digest frequency
+        """
         self.time_diff = FREQUENCY_DICT[frequency]
         self.curr_time = datetime.datetime.now()
-        self.alert_config_cache = dict()
-        self.kpi_cache = dict()
+        self.alert_config_cache: Dict[int, Alert] = dict()
```

Comment:
It's not needed when the value assigned to it has a good type. `FREQUENCY_DICT` here is `Dict[str, timedelta]` and since `frequency` is `str`, it can infer that `time_diff` is `timedelta`.

Reply:
Thanks for clarification !